chia/consensus/blockchain.py

```diff
@@ -657,18 +659,23 @@ async def get_header_blocks_in_range(self, start: int, stop: int) -> Dict[bytes3
         for block in blocks:
             if self.height_to_hash(block.height) != block.header_hash:
                 raise ValueError(f"Block at {block.header_hash} is no longer in the blockchain (it's in a fork)")
-            tx_additions: List[CoinRecord] = [
-                c for c in (await self.coin_store.get_coins_added_at_height(block.height)) if not c.coinbase
-            ]
-            removed: List[CoinRecord] = await self.coin_store.get_coins_removed_at_height(block.height)
-            header = get_block_header(
-                block, [record.coin for record in tx_additions], [record.coin.name() for record in removed]
-            )
+            if no_tx_filter:
```

Comment:
it's the call to `get_blocks_by_hash()` that's really expensive, so we would still call that unconditionally. I wouldn't expect this to have a material impact. At least not on the CPU hotspots I've measured.

Reply:
thse are times for pulling 800 blocks from mainnet db
without tx filter 15:53:21 1.6693732738494873
with tx filter 15:54:07 46.016254901885986